mitlib-cf7-elements.php

```diff
@@ -50,25 +49,44 @@ function add_custom_elements() {
  * @param object $tag    A WPCF7_FormTag object.
  * @link https://contactform7.com/2015/03/28/custom-validation/
  */
-function validate_dlc_filter( $result, $tag ) {
-	// Do the work of validation.
+function identify_required( $result, $tag ) {
+	// Check if the field is marked as required.
 	if ( 'select_dlc*' == $tag->type ) {
-		$value = isset( $_POST['department'] ) ? trim( $_POST['department'] ) : '';
+		$result = validate_dlc_filter( $result, $tag );
+	}
+	return $result;
+}
+add_filter( 'wpcf7_validate_select_dlc*', 'identify_required', 20, 2 );
 
-		if ( '' == $value ) {
-			$result->invalidate( $tag, 'Please specify your department, lab, or center.' );
-		}
+
+/**
+ * Implements custom validation for DLC selection.
+ *
+ * @param object $result A WPCF7_Validation object.
+ * @param object $tag    A WPCF7_FormTag object.
+ * @link https://contactform7.com/2015/03/28/custom-validation/
+ */
+function validate_dlc_filter( $result, $tag ) {
+	// Check for nonce.
+	if ( empty( $_POST['_wpnonce'] ) || ! wp_verify_nonce( sanitize_key( $_POST['_wpnonce'] ), 'wp_rest' ) ) {
```

Comment:
I'd remove the nonce check. If the request is not authenticated then the nonce doesn't provide any protection.

Reply:
To be clear, if I take out this check I'll most likely have to stop PHPCS from checking for that type of code violation (which I can do, but there's no way to selectively exclude a check - either the entire codebase is checked, or it isn't).